blocklib/blocks/null_source/null_source_cpu.cc

```diff
@@ -14,16 +14,15 @@ namespace blocks {
 
 null_source_cpu::null_source_cpu(block_args args) : INHERITED_CONSTRUCTORS {}
 
-work_return_code_t null_source_cpu::work(std::vector<block_work_input_sptr>& work_input,
-                                         std::vector<block_work_output_sptr>& work_output)
+work_return_code_t null_source_cpu::work(work_io& wio)
 {
+    auto itemsize = wio.output(0)->buffer->item_size();
     void* optr;
-    auto itemsize = work_output[0]->buffer->item_size();
-    for (size_t n = 0; n < work_output.size(); n++) {
-        optr = work_output[n]->items<void>();
-        auto noutput_items = work_output[n]->n_items;
+    for (size_t n = 0; n < wio.nout(); n++) {
```

Comment:
would maybe look nicer as `for(auto& output : wio.outputs) { output-> n_produced = noutput_items; }`